### PR TITLE
added dockerfile for pdata 1.28.1; corrected previous pangolin READMEs

### PR DESCRIPTION
**Pull request by kapsakcj**
Simply updated the pangolin data version:

```bash
$ diff pangolin/4.3.1-pdata-1.28/Dockerfile pangolin/4.3.1-pdata-1.28.1/Dockerfile 
12c12
< ARG PANGOLIN_DATA_VER="v1.28"
---
> ARG PANGOLIN_DATA_VER="v1.28.1"
```

I also corrected a few spots in older Pangolin READMEs that had the incorrect pangolin-data version listed.

P-data 1.28.1 release notes here: https://github.com/cov-lineages/pangolin-data/releases/tag/v1.28.1

> _Removed annotation of KR.2 which was accidentally included in v1.28 as a child of LT.1 instead of being completely replaced by LT.1, causing most LT.1 sequences to be assigned KR.2. (See https://github.com/cov-lineages/pangolin-data/issues/60) No change to alias file, still pango-designation v1.28._

This version release addresses this bug ^

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: pangolin/4.3.1-pdata-1.28.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+# modify environment.yml to pin specific versions during install
+# pin specific versions of usher, scorpio, pangolin-data, constellations, and pulp
+# create the conda environment using modified environment.yml
+RUN sed -i "s|usher.*|usher=${USHER_VER}|" /pangolin/environment.yml && \
+ sed -i "s|scorpio.git|scorpio.git@${SCORPIO_VER}|" /pangolin/environment.yml && \
+ sed -i "s|pangolin-data.git|pangolin-data.git@${PANGOLIN_DATA_VER}|" /pangolin/environment.yml && \
+ sed -i "s|constellations.git|constellations.git@${CONSTELLATIONS_VER}|" /pangolin/environment.yml && \
+ sed -i "12 a\  - pulp=2.7.0" /pangolin/environment.yml && \
+ micromamba create -n pangolin -y -f /pangolin/environment.yml
```
[the remark this answers is not in the document]
Comment (kapsakcj) [reply pangolin/4.3.1-pdata-1.28.1/Dockerfile:56]: the command `micromamba clean -a -y` is already on line 69

I tried adding the `-f` flag, but it didn't make a difference in image size:
```
$ docker images | grep 'AFY'
kapsakcj/pangolin     4.3.1-pdata-1.28.1-WithCleanAFY          71b48291a521   14 minutes ago   2.76GB
kapsakcj/pangolin     4.3.1-pdata-1.28.1-WithoutCleanAFY       a2e7869f6fb8   4 days ago       2.76GB
```

I'll add it in there anyways as we're doing this with other dockerfiles. Just want to test the build once more and I'll push a commit

[the remark this answers is not in the document]
Comment (Kincekara) [reply pangolin/4.3.1-pdata-1.28.1/Dockerfile:56]: @kapsakcj It should be on the same layer, just after `micromamba create -n pangolin -y -f /pangolin/environment.yml`.
There is a considerable difference in size.

``` $ docker images | grep clean
ck_clean_afy   latest    427afeda3c0c   37 seconds ago   2.76GB
clean_afy      latest    ab269df21352   5 minutes ago    1.8GB

[the remark this answers is not in the document]
Comment (kapsakcj) [reply pangolin/4.3.1-pdata-1.28.1/Dockerfile:56]: oh wow OK, thanks. I'll try that out

**Comment by kapsakcj** [comment]
I made the suggested change - tests succeeded as expected and the docker image is dramatically smaller. TY for the suggestion Kuluhan & Erin 👍 

OK to merge and deploy? I would suggest using the image tag `4.3.1-pdata-1.28.1` to keep with our naming scheme. (and overwrite the `latest` tag)

**Review by Kincekara — APPROVED** [review]

**Comment by Kincekara** [comment]
Thank you, Curtis! The tests look good! I will merge and deploy this.

**Comment by Kincekara** [comment]
Deploy link: https://github.com/StaPH-B/docker-builds/actions/runs/9861690475

**Comment by kapsakcj** [comment]
TY!

**Comment by kapsakcj** [comment]
check out the (compressed) image size difference:
image

**Pull request merged**